### PR TITLE
[core] Move out RNG impl into specialized files

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

* Remove random_device from RNG seeding
* Add PCG32 RNG & PCG64 that can be uncommented to be used instead of Mersenne Twister.

Default RNG is still Mersenne Twister 32 bit
## Steps to test these changes

play game, see random numbers

### CODE REVIEW

**Comment by zach2good** [comment]
Do we have to be importing all the content of these new libs? Can we download them at configure time using CPM instead?

**Comment by zach2good** [comment]
Can RNG folder become rng?

**Comment by WinterSolstice8** [comment]
CPM downloads these to your build directory, but I can trim the fat quite a bit

**Comment by WinterSolstice8** [comment]
CPM now downloads the repo correctly

File: src/common/xirand.h [hunk 1]
```diff
@@ -23,119 +23,118 @@
...
-            *it ^= std::chrono::duration_cast<std::chrono::microseconds>(
-                       std::chrono::high_resolution_clock::now().time_since_epoch())
-                       .count();
-
-            //*it ^= more_external_random_stuff;
-        }
-        std::seed_seq seq(seed_data.cbegin(), seed_data.cend());
-        mt().seed(seq);
-    }
+// You can choose an RNG by commenting/uncommenting this line. The default is Mersenne Twister in 32 bit.
```
Comment (zach2good) [inline src/common/xirand.h:26]: 👍

**Review by zach2good — APPROVED** [review]

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**